Checkout/Checkout/Source/Tokenisation/TokenDetailsFactory.swift

```diff
@@ -18,7 +18,7 @@ final class TokenDetailsFactory: TokenDetailsProviding {
       token: tokenResponse.token,
       expiresOn: tokenResponse.expiresOn,
       expiryDate: create(expiryMonth: tokenResponse.expiryMonth, expiryYear: tokenResponse.expiryYear),
-      scheme: tokenResponse.scheme.flatMap(Card.Scheme.init(rawValue:)),
+      scheme: tokenResponse.scheme.flatMap({ Card.Scheme(rawValue: $0) }),
```

Comment:
what is the purpose of this change? Function reference is cleaner IMO

Reply:
The syntax wouldn't compile. I think it worked before when rawValue was coming from compiler, whereas now we're mimicking that behaviour to reduce number of small syntax changes (and because its a familiar pattern), but ultimately Card.Scheme no longer has an actual String rawValue as far as compiler is concerned (associated value side effect)

<img width="520" alt="Screenshot 2022-08-11 at 14 41 10" src="https://user-images.githubusercontent.com/102028170/184147114-1327762b-adb5-4a95-ab73-b27949e1c7aa.png">

Reply:
Looked into this and its because of the default for the length. The init is `(String, Int) -> Card.Scheme?`

Reply:
IDE is not very supportive, I'm not sure the syntax could be made to compile? In my book I am happy to use either approach, but only one allows for sufficient brevity and links well in this case

Reply:
What you have already is the cleanest way of doing this